python/ml4ir/applications/ranking/tests/test_feature_layer.py

```diff
@@ -282,10 +282,80 @@ def test_categorical_embedding_with_vocabulary_file(self):
         assert not tf.reduce_all(tf.equal(categorical_embedding[0], categorical_embedding[1]))
         assert tf.reduce_all(tf.equal(categorical_embedding[4], categorical_embedding[5]))
 
-        # Strings group_0 and group_2 should NOT result in the same embedding because they use a default one-to-one vocabulary mapping
+        # Strings domain_0 and domain_2 should NOT result in the same embedding because they use a default one-to-one vocabulary mapping
         assert not tf.reduce_all(tf.equal(categorical_embedding[0], categorical_embedding[3]))
         assert not tf.reduce_all(tf.equal(categorical_embedding[3], categorical_embedding[4]))
 
+    def test_categorical_embedding_with_vocabulary_file_and_dropout(self):
+        """
+        Asserts the conversion of a categorical string tensor into an embedding representation
+        Works by converting the string into indices using a vocabulary file and then dropping these indices into the OOV index at dropout_rate rate.
+
+        The embedding size, dropout_rate are controlled by feature_info
+        """
+        #####################################################
+        # Test for vocabulary file with ids mapping specified
+        #####################################################
+        embedding_size = 4
+        dropout_rate = 0.999
+        feature_info = {
+            "name": "categorical_variable",
+            "feature_layer_info": {
+                "fn": "categorical_embedding_with_vocabulary_file",
+                "args": {
+                    "vocabulary_file": "ml4ir/applications/ranking/tests/data/config/domain_name_vocab.csv",
+                    "embedding_size": embedding_size,
+                    "dropout_rate": dropout_rate,
+                },
+            },
+        }
+
+        # Define an input string tensor
+        string_tensor = tf.constant(
+            ["domain_0", "domain_1", "domain_0", "domain_2", "domain_10", "domain_11"]
+        )
+        try:
+            categorical_embedding = categorical_fns.categorical_embedding_with_vocabulary_file_and_dropout(
+                string_tensor, feature_info, self.file_io
+            )
+        except ValueError:
+            # Should throw error as method does not work with IDs containing 0
+            assert True
+
+        ########################################################
+        # Test for vocabulary file with no ids mapping specified
```

Comment:
But it's still the same code path, right?

Reply:
Yes, but it helps to not test everything in a single test.
- So that when a test fail, we know why (the reason is limited), and can segment more easily the issue.
- Here, we won't know if a change breaks everything after the first failure.
- It help from a CI perspective to easily track the amount of failures.
- And it's good practice :)

Reply:
Here the test could fail for 10 different asserts.

Reply:
I will break it up.